### PR TITLE
add experimental `useSuspenseQuery()`

### LINKED ISSUE #1354: Make `useQuery` with `suspense: true` to return non-nullable `data`
### Describe the feature you'd like to request

`{ suspense: true }` should let us use `data` without checking its integrity.

We can use `suspense: true` in SSR now, see #3130

### Describe the solution you'd like to see

Given

```ts
const postQuery = trpc.post.byId.useQuery({ id }, { suspense: true });
```

`postQuery.data` should be **not** be nullable.

Considerations: if `enabled: true`, we should not infer it as non-nullable

### Desribe alternate solutions

- Default options to set `suspense: true`
- Add a `useSuspenseQuery` / `useSuspenseInfiniteQuery`

Another approach would be to enable this behavior globally when creating the react-query hooks.

```tsx
export trpc = createTRPCReact<Context>({suspense: true})
```

### Additional information

- `enabled:` `true`/`false` should still work
- We need to narrow the generic so we still can't pass extra params to `opts`

### PR DESCRIPTION
**Pull request by KATT**
Closes #1354

## 🎯 Changes

- Adds experimental `useSuspenseQuery()` 
- It is **feature-flagged** as the exact implementation might change & we're waiting to see how the `react-query` team on how they're planning to deal with it
- Live demo: https://github.com/KATT/useSuspense-next-prisma-starter

### Enable experimental suspense

```tsx
const trpc = createTRPCReact<
  AppRouter,
  unknown,
  'ExperimentalSuspense' // <------------- This is the magical feature flag
>();
```

### Using it

The above will give you access to a new fresh hook:

```ts
const [data] = trpc.post.byId.useSuspenseQuery({ id: "1" })
//      ^? will be typeeeed and no loading state stuff
```

## ✅ Checklist

- [x] I have followed the steps listed in the [Contributing guide](https://github.com/trpc/trpc/blob/next/CONTRIBUTING.md).
- [ ] If necessary, I have added documentation related to the changes made.
- [x] I have added or updated the tests related to the changes made.

### CODE REVIEW

File: packages/tests/server/react/__reactHelpers.tsx [hunk 1]
```diff
@@ -36,7 +36,11 @@ export function getServerAndReactClient<TRouter extends AnyRouter>(
   });
 
   const queryClient = createQueryClient();
-  const proxy = createTRPCReact<typeof appRouter>();
+  const proxy = createTRPCReact<
+    typeof appRouter,
+    unknown,
+    'ExperimentalSuspense'
+  >();
```
Comment (KATT) [inline packages/tests/server/react/__reactHelpers.tsx:43]: This is how you enable it

Comment (sachinraja) [reply packages/tests/server/react/__reactHelpers.tsx:43]: should feature flags be an array?

Comment (KATT) [reply packages/tests/server/react/__reactHelpers.tsx:43]: I'd also want autocomplete on them. Types became annoying. Feel free to have a stab at improving it

Comment (KATT) [reply packages/tests/server/react/__reactHelpers.tsx:43]: It doesn't need to be an array, it could work like the second arg of `Pick` etc

Comment (sachinraja) [reply packages/tests/server/react/__reactHelpers.tsx:43]: Ahh I read the `'ExperimentalSuspense'` argument as runtime code the first time but it's passed as a generic

File: packages/next/src/createTRPCNext.tsx [hunk 2]
```diff
@@ -14,6 +14,7 @@ import { WithTRPCNoSSROptions, WithTRPCSSROptions, withTRPC } from './withTRPC';
 export function createTRPCNext<
   TRouter extends AnyRouter,
   TSSRContext extends NextPageContext = NextPageContext,
+  TFlags = null,
```
Comment (sachinraja) [inline packages/next/src/createTRPCNext.tsx:17]:
```suggestion
  TFlags = never,
```

**Pull request merged**